cyclone/R/prepare_checkpoint1.R

```diff
@@ -92,6 +109,31 @@ prepare_checkpoint1 <- function(raw_exp=NULL, trans_exp=NULL, file_metadata, mar
     )
   }
   
+  # Handle cell/sample-use parameters
+  if (exclude_controls || subsample) {
+    # Handle control exclusion
+    if (exclude_controls) {
+      file_metadata <- file_metadata %>% filter( ! control_sample )
+      cell_metadata <- cell_metadata %>% filter( ! control_sample )
+    }
+    # Subsampling if subsample == TRUE
+    if (subsample) {
+      cells_use <- unlist(lapply(
+        file_metadata$file_name,
+        function(file) {
+          cells_in_sample <- rownames(cell_metadata)[cell_metadata$file_name==file]
+          sample(cells_in_sample, min(subsample_n, length(cells_in_sample)), replace = FALSE)
```

Comment:
I agree that this is a good idea since we're targeting all skill levels.

But also wanna note that it's something we'd be dinged for if we ever try to submit cyclone to Bioconductor.  If we do bring cyclone to that point, we'll want to suggest setting a seed in example/vignette code instead, and remove from the function itself.

I'll add it in with an input to give seed / ignorance control.

Reply:
That's a good idea!

Reply:
added.  if you approve the new commit, feel free to merge this PR!

Reply:
I like the `if` clause for setting the seed! :D